backend/src/xfd_django/xfd_api/auth_saml.py

```diff
@@ -0,0 +1,362 @@
+"""Auth SAML routes for FastAPI backend."""
+
+# Standard Python Libraries
+from datetime import datetime, timezone
+import logging
+import os
+from typing import Any, Dict, Optional
+import urllib.parse
+
+# Third-Party Libraries
+from django.conf import settings as dj_settings
+from fastapi import APIRouter, HTTPException, Request
+from fastapi.responses import RedirectResponse, Response
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser
+from onelogin.saml2.settings import OneLogin_Saml2_Settings
+from xfd_mini_dl.models import User
+
+from .auth import (
+    create_jwt_token,
+    update_login_block_status,
+    user_to_dict,
+    validate_json_serialization,
+)
+
+logging.basicConfig(level=logging.INFO)
```

Comment:
I think there is already a basicConfig in settings.py. In practice, one basicConfig is desirable in Django, if you have more than one, can cause unexpected results.